clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -318,4 +318,11 @@ public AttributeList setAttributes(AttributeList list) {
                                       + ".(whitelist/blacklist) is not a valid regular expression");
         }
     }
+
+    @Override
+    public synchronized void contextChange(MetricsContext metricsContext) {
+        if (this.prefix == null || this.prefix.isEmpty()) {
```

Comment:
I don't think there is a legitimate case for prefix to ever be `null`. We can probably add a check to the constructor to ensure that's never the case.

Reply:
Check added at constructor.